hipcheck/src/main.rs

```diff
@@ -144,6 +145,130 @@ fn cmd_schema(args: &SchemaArgs) {
+	let src: PathBuf = from.as_ref().to_path_buf();
+	if !src.is_dir() {
+		return Err(hc_error!("source path must be a directory"));
+	}
+	let dst: PathBuf = to.as_ref().to_path_buf();
+	if !dst.is_dir() {
+		return Err(hc_error!("target path must be a directory"));
+	}
+
+	for entry in walkdir::WalkDir::new(&src) {
+		let src_f_path = entry?.path().to_path_buf();
+		if src_f_path == src {
+			continue;
+		}
+		let mut dst_f_path = dst.clone();
+		dst_f_path.push(
+			src_f_path
+				.file_name()
+				.ok_or(hc_error!("src dir entry without file name"))?,
+		);
+		// This is ok for now because we only copy files, no dirs
+		std::fs::copy(src_f_path, dst_f_path)?;
+	}
+
+	Ok(())
+}
+
+fn cmd_setup(config: &CliConfig) -> ExitCode {
+	// Make config dir if not exist
+	let Some(tgt_conf_path) = config.config() else {
+		print_error(&hc_error!("target config dir not specified"));
+		return ExitCode::FAILURE;
+	};
+	if !tgt_conf_path.exists() && create_dir_all(tgt_conf_path).is_err() {
+		print_error(&hc_error!("failed to create missing target config dir"));
+	}
+	let Ok(abs_conf_path) = tgt_conf_path.canonicalize() else {
+		print_error(&hc_error!("failed to canonicalize HC_CONFIG path"));
+		return ExitCode::FAILURE;
+	};
+
+	// Make data dir if not exist
+	let Some(tgt_data_path) = config.data() else {
+		print_error(&hc_error!("target data dir not specified"));
+		return ExitCode::FAILURE;
+	};
+	if !tgt_data_path.exists() && create_dir_all(tgt_data_path).is_err() {
+		print_error(&hc_error!("failed to create missing target data dir"));
+	}
+	let Ok(abs_data_path) = tgt_data_path.canonicalize() else {
+		print_error(&hc_error!("failed to canonicalize HC_DATA path"));
+		return ExitCode::FAILURE;
+	};
+
+	// Copy current `hc` binary to OS-specific bin path if not there already
+	if which::which("hc").is_err() {
+		// Try env var first, then default to OS-specific exec dir
+		let mut tgt_bin_path: PathBuf = if let Ok(bp) = std::env::var("HC_BIN") {
+			PathBuf::from(bp)
+		} else if let Some(bp) = dirs::executable_dir() {
+			bp
+		} else {
+			print_error(&hc_error!("could not find appropriate bin dir for `hc`"));
+			return ExitCode::FAILURE;
+		};
+		if !tgt_bin_path.exists() && create_dir_all(&tgt_bin_path).is_err() {
+			print_error(&hc_error!("failed to create missing HC_BIN dir"));
+		}
+		tgt_bin_path.push("hc");
+		if let Err(e) = std::fs::copy(std::env::current_exe().unwrap(), &tgt_bin_path) {
+			print_error(&hc_error!(
+				"could not find appropriate bin dir for `hc`: {}",
+				e
+			));
+			return ExitCode::FAILURE;
+		}
+		let Ok(abs_bin_path) = tgt_bin_path.canonicalize() else {
+			print_error(&hc_error!("failed to canonicalize HC_BIN path"));
+			return ExitCode::FAILURE;
+		};
+		if which::which("hc").is_err() {
+			println!(
+				"warning: `hc` copied to '{}', which is not in your system path. Consider adding it",
+				abs_bin_path.parent().unwrap().display(),
+			);
+		}
+	}
+
+	// Copy local config/data dirs to target locations
```

Comment:
I know it's not testable until `cargo-dist` has published a new version, but the right flow here really will be to get the additional files from the `cargo-dist` bundle for the release, download them, and place them into the correct location.

Reply:
Would you rather leave this functionality out for now, or keep/alter it and make a ticket to update it once we have a compatible `cargo-dist` bundle?

Reply:
I think this MR will have to wait until I can publish a new release which includes those files with `cargo-dist`.